### PR TITLE
Feature/support auth anonymously

### PR DESCRIPTION
**Pull request by maxis1718**
Use anonymous Connection when given empty `getAuthToken` function

### CODE REVIEW

File: src/Connection.js [hunk 1]
```diff
@@ -10,25 +10,35 @@ let Connection = function(endPoint, getAuthToken) {
     if (!conn) {
       conn = new FB(endPoint)
     }
-
     if ( shouldAuth() ) {
       authConnection()
+    } else {
+      anonymousConnection()
```
Comment (IanWang) [inline src/Connection.js]: please add a test for this

**Comment by maxis1718** [comment]
@IanWang  I've refactored and add more test cases. Please help review again 🙏

**Comment by maxis1718** [comment]
@mz026 Thanks for pointing out the auth logic bug xD Please help review again ✌️

File: src/Connection.js [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
+/*
+ * @param {String} endPoint
+ * @param {Function} [getAuthToken]
+ */
+const Connection = function(endPoint, getAuthToken) {
   let conn
   let authed = false
   let authorizing = false
   let expiresAt = 0
+  const isAnonymous = !getAuthToken
```
Comment (mz026) [inline src/Connection.js]: How about making this an argument explicitly?

**Pull request merged**